clippy_utils/src/check_proc_macro.rs

```diff
@@ -125,6 +121,7 @@ fn qpath_search_pat(path: &QPath<'_>) -> (Pat, Pat) {
 fn expr_search_pat(tcx: TyCtxt<'_>, e: &Expr<'_>) -> (Pat, Pat) {
     match e.kind {
         ExprKind::ConstBlock(_) => (Pat::Str("const"), Pat::Str("}")),
+        // Parenthesis are skipped before the patterns are matched.
```

Comment:
This comment doesn't make much sense to me. I recall expressions within `ExprKind::Paren` retain the parenthesis' span? Can you elaborate on this further? Is this special for `Tup`?

Reply:
It's referring to how the search patterns are matched to the source text. It's why the head and tail patterns for an empty tuple appear flipped.

Reply:
I think I understand now. Can you update this comment then? Perhaps:
```rust
// Parentheses are skipped before the patterns are matched. With an empty tuple,